modules/ossm-vs-istio.adoc

```diff
@@ -29,6 +29,27 @@ The upstream Istio community installation automatically injects the sidecar into
 
 {ProductName} does not automatically inject the sidecar to any pods, but requires you to opt in to injection using an annotation without labeling projects. This method requires fewer privileges and does not conflict with other OpenShift capabilities such as builder pods. To enable automatic injection you specify the `sidecar.istio.io/inject` annotation as described in the Automatic sidecar injection section.
 
+.Sidecar injection label and annotation settings
+[options="header"]
+[cols="a, a, a"]
+|===
+|
+|Upstream Istio
+|{ProductName}
+
+|Namespace Label
+|supports "enabled" and "disabled"
+|not supported
+
+|Pod Label
+|supports "true" and "false"
+|not supported
+
+|Pod Annotation
+|supports "false" only
+|"true" and "false"
+|===
+
```

Comment:
This is how I saw OSSM 2.1 work last November - specifically, that upstream Istio supports auto-injection on a per-namespace basis (using the namespace label) whereas OSSM only supports allowing auto-injection on pods only (using the pod annotation only).

It has come to my attention that this may no longer be the case in OSSM. We should confirm that this is how it is supposed to be. It could be that OSSM did not intend to allow for namespace auto-injection label to take effect but a bug has enabled it. This is just a guess. But I was under the impression OSSM was not allowing namespace auto-injection labels to be supported for security purposes.

So, we should at least get confirmation as to what is the expected behavior of OSSM 2.2. Is it:

(a) OSSM supports auto-injection on a per-namespace basis (via namespace label)

or

(b) OSSM does NOT support auto-injection per-namespace but ONLY supports auto-injection on a per-pod basis (via the pod annotation, and perhaps via a pod label, too?)

Reply:
I'm commenting out this table so that we can move forward while waiting on an answer to the question of what we support.